### DISCUSSION TITLE
How to target a timer running on a Google nest speaker? (As a new timer ID is generated each time)

### DISCUSSION THREAD

**Discussion by Sambuccashake** [leikoilja/ha-google-home#646] [Q&A] [answered]
Hello!

How do I get around the fact that the timer ID is generated upon creation of the timer itself?

Long story short, any created automation (which holds whatever timer_ID I could dig up at the time) will becomes irrelevant the next time a timer is started as the timer_id attribute will be changed into something else.

Which leads to: 

> 
> This error originated from a custom integration.
> 
> Logger: custom_components.google_home
> Source: custom_components/google_home/sensor.py:305
> Integration: Google Home (documentation, issues)
> First occurred: January 31, 2023 at 17:31:37 (18 occurrences)
> Last logged: 11:06:31
> 
> Incorrect ID format! Please provide a valid timer ID. See services tab for more info.

Do you have any suggestion on how to catch them "dynamically"?

Thanks in advance!

**Comment by KapJI** [comment, marked as answer]
You can use templates (same can be done with timers):
```jinja
{% set alarms = state_attr("sensor.bedroom_speaker_alarms", "alarms") %}
Alarm ID: {{ "Not set" if not alarms else alarms[0]["alarm_id"] }} 
```

**Reply by Sambuccashake** [reply]
Thanks for helping out! I've never dabbled with templates so will have to read up on it.

But not really sure how it will solve my predicament.

`{% set timers = state_attr("sensor.koket_timers", "timer") %}
Timer ID: {{ "Not set" if not timers else timers[0]["timer_id"] }} `

My automation is pretty straight forward and if I could just get past that timer_ID (And just turn any timer off on the device) it would be done but that seems mandatory info.

`alias: Kill timer
description: ""
trigger:
  - device_id: 3e4571e4038a0a51243c40977a96af65
    domain: zha
    platform: device
    type: remote_button_short_press
    subtype: button
condition: []
action:
  - service: google_home.delete_timer
    data:
      entity_id: sensor.koket_timers
      timer_id: timer/63e46722-0000-287b-86a1-14223bb8f172
mode: single`
